aten/src/ATen/native/quantized/cpu/ACLUtils.h

```diff
@@ -0,0 +1,257 @@
+        // PR: https://review.mlplatform.org/c/ml/ComputeLibrary/+/12820/8 This
+        // will be set to the actual src offset value at runtime.
+        arm_compute::QuantizationInfo(
+            1.0, std::numeric_limits<int8_t>::max(), true));
+    acl_gemm->src_s8_tensor_info.set_are_values_constant(false);
+
+    acl_gemm->wei_tensor_info = arm_compute::TensorInfo(
+        arm_compute::TensorShape(n_, k_),
+        1,
+        arm_compute::DataType::QASYMM8_SIGNED,
+        arm_compute::QuantizationInfo(wei_scale_, wei_zero_point_, true));
+    acl_gemm->wei_tensor_info.set_are_values_constant(true);
+
+    // True iff the linear layer has bias && all std::optional bias containers
+    // have a value
+    bool with_bias{false};
+    if (bias_.has_value()) {
+      acl_gemm->bia_tensor_info = arm_compute::TensorInfo(
+          arm_compute::TensorShape(1, n_), 1, arm_compute::DataType::F32);
+      acl_gemm->bia_tensor = arm_compute::Tensor();
+      with_bias = true;
+    }
+
+    acl_gemm->dst_tensor_info = arm_compute::TensorInfo(
+        arm_compute::TensorShape(n_, m), arm_compute::Format::F32);
+
+    // validate that ACL can handle the given problem and inputs.
+    if (fuse_relu) {
+      arm_compute::Status relu_status =
+          arm_compute::NEActivationLayer::validate(
+              &acl_gemm->dst_tensor_info,
+              &acl_gemm->dst_tensor_info,
+              acl_gemm->acl_relu_info);
+      if (relu_status.error_code() != arm_compute::ErrorCode::OK) {
+        return nullptr;
+      }
+    }
+    arm_compute::Status quant_status =
+        arm_compute::NEQuantizationLayer::validate(
+            &acl_gemm->src_fp32_tensor_info, &acl_gemm->src_s8_tensor_info);
+    if (quant_status.error_code() != arm_compute::ErrorCode::OK) {
+      return nullptr;
+    }
+    arm_compute::Status gemm_status =
+        arm_compute::NEGEMMLowpMatrixMultiplyCore::validate(
+            &acl_gemm->src_s8_tensor_info,
+            &acl_gemm->wei_tensor_info,
+            with_bias ? &acl_gemm->bia_tensor_info.value() : nullptr,
+            &acl_gemm->dst_tensor_info,
+            acl_gemm->gemm_info);
+
+    if (gemm_status.error_code() != arm_compute::ErrorCode::OK) {
+      return nullptr;
+    }
+
+    // set the tensor info (i.e. shape, datatype, quant info) for the ACL
+    // tensors
+    acl_gemm->src_fp32_tensor.allocator()->init(acl_gemm->src_fp32_tensor_info);
+    acl_gemm->src_s8_tensor.allocator()->init(acl_gemm->src_s8_tensor_info);
+    acl_gemm->wei_tensor.allocator()->init(acl_gemm->wei_tensor_info);
+    if (with_bias) {
+      acl_gemm->bia_tensor.value().allocator()->init(
+          acl_gemm->bia_tensor_info.value());
+    }
+    acl_gemm->dst_tensor.allocator()->init(acl_gemm->dst_tensor_info);
+
+    // allocate memory only for the quantized tensor, the rest will use memory
+    // already avaliable from PyTorch
+    acl_gemm->src_s8_tensor.allocator()->allocate();
```

Comment:
Just curious, why use ACL allocator instead of an existing PyTorch one? So that memory tracking story is cleaner, i.e. all memory is allocated/tracked and freed by PyTorch caching allocator?

Reply:
I agree with you. 
I now use PyTorch for all allocations - ACL just import pointers but does not explicitly allocate/deallocate any memory.
See [here](https://github.com/pytorch/pytorch/pull/148585/files#diff-db9f299d73216de73f5f8efb40022111c1e26eec780e49dbe70439cb7249013aR104)